core/src/main/java/org/apache/iceberg/view/BaseMetastoreViewCatalog.java

```diff
@@ -67,6 +71,31 @@ public View loadView(TableIdentifier identifier) {
     throw new NoSuchViewException("Invalid view identifier: %s", identifier);
   }
 
+  @Override
+  public Table loadTable(TableIdentifier identifier) {
+    try {
+      return super.loadTable(identifier);
+    } catch (NoSuchTableException e) {
+      return loadViewMetadataTable(identifier);
+    }
+  }
+
+  private Table loadViewMetadataTable(TableIdentifier identifier) {
+    String tableName = identifier.name();
+    ViewMetadataTableType type = ViewMetadataTableType.from(tableName);
+    if (type != null) {
+      TableIdentifier baseViewIdentifier = TableIdentifier.of(identifier.namespace().levels());
+      ViewOperations ops = newViewOps(baseViewIdentifier);
+      if (ops.current() == null) {
+        throw new NoSuchTableException("Table or View does not exist: %s", baseViewIdentifier);
+      }
+      return MetadataTableUtils.createViewMetadataTableInstance(
+          ops, name(), baseViewIdentifier, identifier, type);
+    } else {
+      throw new NoSuchTableException("Table does not exist: %s", identifier);
```

Comment:
we should probably also throw the original exception that was caught in `loadTable`